### DISCUSSION TITLE
Various problems as a beginner

### DISCUSSION THREAD

**Discussion by MOS5500** [fossi-foundation/open-pdks#451] [Q&A]
Ok, this is not a real problem, but a series of questions from a newbie.
I don't know where to turn, if you think it's wrong to post them here, I apologize but please, let me know where I can post them.
First of all, after a lot of troubles, I apparently managed to download and make the Open PDK and some tools work. I'm a newbie, I have knowledge of electronics (even if I'm a computer scientist) but not at such a low level. It's the first time I try to create something concrete that goes beyond university exercises.
1) For the open pdk and skywater community, is there an official forum where you can write or share your experiences? Or something on reddit or similar?
2) I made some prototypes by creating schematics with the Skywater components with xschem and I simulated their behavior with ngspice. I'm using custom mosfet formats (W/L). everything works well so far and I would like to transform everything into a layout. I am using Magic.
From the videos on youtibe I see that Magic has two buttons (“Design 1” and “Design 2”) that my Magic configuration does not have. I guess I have configured magic wrong, but I do not know where to find a guide.

3) If I understand correctly, once I have saved the netlist with Xschem, I can import it into magic and automatically get the layout of the components that I used in Xschem. Great! Except that… I cannot load the netlist. From magic I do: readspice myfile.spice, but I get the message: “Annotating port orders from myfile.spice”. What to do? Is this also a configuration problem?
4) In the Skywater documentation I read that the spice models are valid in a certain voltage range (eg. 0..1.8V), but I need to make them work in a different range. Ngspice gives me what I expect even using my range (-1.8v to 1.8v) but I do not know if at this point the simulation is reliable.
5) It's still early to talk about it but... are there any ready-made layouts for the chip pads?

Thank you for your attention and patience.

**Comment by RTimothyEdwards** [comment]
1. Yes, there is a forum!  Go to https://efabless.com and click on the Slack icon at the top.
If that doesn't get you the registration link for the Slack workspace, then email me (tim@efabless.com) and I can add you.'

2. To get magic to show the "Devices 1" and "Devices 2" menus, assuming that you got open_pdks compiled and installed, make sure you start magic with (assuming a default install location of /usr/local/share/pdk):  `magic -rcfile /usr/local/share/pdk/sky130A/libs.tech/magic/sky130A.magicrc`.  The "magicrc" startup script loads the Tcl script that defines the generated devices and adds the two drop-down menus.

3. The "readspice" command is for annotating ports (as you found).  There is an "Import SPICE" menu item which is what you are looking for.  Like "Devices 1" and "Devices 2", it is created when you start magic with the right startup script for the PDK.

4. Generally, CMOS processes don't use negative voltages, and while it is likely _possible_ to set up a circuit in Sky130 to use negative voltages, by far the more common setup would be to use a 0 to 3.3 (or 3.6, if you must) supply and use an amplifier to drive a common-mode ground at the midpoint between ground and supply.  Since all voltages are relative, 0 to 3.6 is equivalent to -1.8 to +1.8, and especially in simulation, it is exactly equivalent.  The issue is really just making sure that your midpoint voltage reference is stable.

5.  Yes, there are ready-made layouts for the chip pads!  There are entire ready-made padframes, and ready-made chips.  See efabless.com for details.  Also there is much discussion about these on the above-mentioned Slack workspace.
